streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamImplTest.java

```diff
@@ -2330,4 +2348,266 @@ public void shouldNotAllowNullNamedOnProcessWithStores() {
+                mkEntry("lhs1", "(lhsValue1|rhs1,rhsValue1)"),
+                mkEntry("lhs2", "(lhsValue2|rhs2,rhsValue2)")
+            );
+            assertThat(outputTopic.readKeyValuesToMap(), is(expected));
+
+            // Add another reference to an existing FK
+            left.pipeInput("lhs3", "lhsValue3|rhs1");
+
+            assertThat(
+                outputTopic.readKeyValuesToMap(),
+                is(mkMap(
+                    mkEntry("lhs3", "(lhsValue3|rhs1,rhsValue1)")
+                ))
+            );
+
+            left.pipeInput("lhs1", (String) null);
+            assertThat(
+                outputTopic.readKeyValuesToMap(),
+                is(mkMap(
+                    mkEntry("lhs1", null)
+                ))
+            );
+        }
+    }
+
+    @Test
+    public void shouldKStreamToKTableInnerJoin() {
+        final Consumed<String, String> consumed = Consumed.with(Serdes.String(), Serdes.String());
+        final StreamsBuilder builder = new StreamsBuilder();
+
+        final String input1 = "input1";
+        final String input2 = "input2";
+        final String output = "output";
+
+        final KTable<String, String> table1 = builder.stream(input1, consumed).toTable();
+
+        final KTable<String, String> table2 = builder.table(input2, consumed);
+        table1.join(table2, MockValueJoiner.TOSTRING_JOINER).toStream().to(output);
+
+        final Collection<Set<String>> copartitionGroups =
+            TopologyWrapper.getInternalTopologyBuilder(builder.build()).copartitionGroups();
+
+        assertEquals(1, copartitionGroups.size());
+        assertEquals(new HashSet<>(Arrays.asList(input1, input2)), copartitionGroups.iterator().next());
+
+        try (final TopologyTestDriver driver = new TopologyTestDriver(builder.build(), props)) {
+            final TestInputTopic<String, String> inputTopic1 =
+                driver.createInputTopic(input1, Serdes.String().serializer(), Serdes.String().serializer(), Instant.ofEpochMilli(0L), Duration.ZERO);
+            final TestInputTopic<String, String> inputTopic2 =
+                driver.createInputTopic(input2, Serdes.String().serializer(), Serdes.String().serializer(), Instant.ofEpochMilli(0L), Duration.ZERO);
+            final TestOutputTopic<String, String> outputTopic =
+                driver.createOutputTopic(output, Serdes.String().deserializer(), Serdes.String().deserializer());
+
+            inputTopic2.pipeInput("1", "stream");
+
+            assertThat(
+                outputTopic.readKeyValuesToMap(),
+                is(emptyMap())
+            );
+
+            inputTopic1.pipeInput("1", "table");
+
+            final Map<String, String> expected = mkMap(
+                mkEntry("1", "table+stream")
+            );
+            assertThat(
+                outputTopic.readKeyValuesToMap(),
+                is(expected)
+            );
+        }
+    }
+
+    @Test
+    public void shouldKStreamToKTableLeftJoin() {
```

Comment:
I am frankly not sure, if we need to test left-table-table join explicitly (the table-table join test from above should be sufficient)